app/waf/src/controllers/rule.controller.ts

```diff
@@ -139,4 +147,239 @@ export class RuleController {
+    return await this.actionRepository.find(filter);
+  }
+
+  @get(prefix + '/conditions/count', {
+    responses: {
+      '200': {
+        description: 'Condition model count',
+        content: {'application/json': {schema: CountSchema}},
+      },
+    },
+  })
+  async countConditions(
+    @param.query.object('where', getWhereSchemaFor(Condition)) where?: Where,
+  ): Promise<Count> {
+    return await this.conditionRepository.count(where);
+  }
+
+  @get(prefix + '/conditions', {
+    responses: {
+      '200': {
+        description: 'Array of Condition model instances',
+        content: {
+          'application/json': {
+            schema: {type: 'array', items: {'x-ts-type': Condition}},
+          },
+        },
+      },
+    },
+  })
+  async findConditions(
+    @param.query.object('filter', getFilterSchemaFor(Condition))
+    filter?: Filter,
+  ): Promise<Condition[]> {
+    return await this.conditionRepository.find(filter);
+  }
+
+  @post(prefix + '/rules/{rule_id}/conditions', {
+    responses: {
+      '200': {
+        description: 'Conditions add to rule success',
+        content: {'application/json': {schema: {'x-ts-type': Condition}}},
+      },
+    },
+  })
+  async createRuleCondition(
+    @param.path.string('rule_id') rule_id: string,
+    @requestBody() condition: Partial<Condition>,
+  ): Promise<Condition> {
+    return await this.ruleRepository.conditions(rule_id).create(condition);
+  }
+
+  @get(prefix + '/rules/{rule_id}/conditions/{condition_id}', {
+    responses: {
+      '200': {
+        description: 'Condition model instance',
+        content: {'application/json': {schema: {'x-ts-type': Condition}}},
+      },
+    },
+  })
+  async getConditionByID(
+    @param.path.string('rule_id') rule_id: string,
+    @param.path.string('condition_id') condition_id: string,
+  ): Promise<Condition> {
+    const result = await this.ruleRepository
+      .conditions(rule_id)
+      .find({where: {id: condition_id}});
+    return result[0];
+  }
+
+  @get(prefix + '/rules/{rule_id}/conditions', {
+    responses: {
+      '200': {
+        description: 'Array of conditon model instances',
+        content: {
+          'application/json': {
+            schema: {type: 'array', items: {'x-ts-type': Condition}},
+          },
+        },
+      },
+    },
+  })
+  async getConditions(
+    @param.path.string('rule_id') rule_id: string,
+  ): Promise<Condition[]> {
+    return await this.ruleRepository.conditions(rule_id).find();
+  }
+
+  @del(prefix + '/rules/{rule_id}/conditions/{condition_id}', {
+    responses: {
+      '204': {
+        description: 'Condition DELETE success',
+      },
+    },
+  })
+  async deleteConditionByID(
+    @param.path.string('rule_id') rule_id: string,
+    @param.path.string('condition_id') condition_id: string,
+  ) {
+    await this.ruleRepository.conditions(rule_id).delete({id: condition_id});
+  }
+
+  @put(prefix + '/rules/{rule_id}/conditions/{condition_id}', {
```

Comment:
As we discussed in the meeting, we will support PATCH API, instead of PUT.

Reply:
Will correct it.